### PR TITLE
Migration 8 

### PR DESCRIPTION
**Pull request by AuHau**

### CODE REVIEW

File: migrations/migration-8/blocks-to-multihash.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+async function process(repoPath, options, keyFunction){
+  const { StorageBackend, storageOptions } = utils.getDatastoreAndOptions(options, 'blocks')
+
+  const baseStore = new StorageBackend(path.join(repoPath, 'blocks'), storageOptions)
+  const store = await maybeWithSharding(baseStore, storageOptions)
+
+  try {
+    const batch = store.batch()
```
Comment (alanshaw) [inline migrations/migration-8/blocks-to-multihash.js]: We can't buffer the entire contents of a repo into memory. This has to be done incrementally.

**Review by alanshaw — CHANGES_REQUESTED** [review]
Do you think that we should move `migrations` to `src/migrations`?

Can you explain for me the need for `keys-encoding.js`?

**Comment by AuHau** [comment]
Tbh I like having the `migrations` separate from `src`. The `src` contains the migration's framework which the `migrations` does not directly belong to.

Regarding the `key-encoding` this is related to https://github.com/ipfs/js-ipfs/issues/1937 but as mentioned in related PR https://github.com/ipfs/js-ipfs/pull/2410 I am not sure if it is needed (please see the PR and comment there if possible)

**Comment by achingbrain** [comment]
Tests are failing in the browser due to https://github.com/ipfs/js-stores/issues/198

**Comment by achingbrain** [comment]
~Depends on:~

~- [ ] https://github.com/ipfs/js-datastore-idb/pull/7~

Not any more - we've switched back to `datastore-level` in the browser until https://github.com/ipfs/js-stores/issues/198 is resolved.

**Pull request merged**